evap/evaluation/tests/test_views.py

```diff
@@ -246,3 +248,44 @@ def test_notebook(self):
 
         user.refresh_from_db()
         self.assertEqual(user.notes, self.note)
+
+
+class TestResetEvaluation(WebTestStaffMode):
+
+    @classmethod
+    def setUpTestData(cls) -> None:
+        cls.manager = make_manager()
+        cls.semester = baker.make(Semester, results_are_archived=True)
+
+    def reset_from_x_to_new(self, x, success_expected: bool) -> None:
+        evaluation = baker.make(Evaluation, state=x, course__semester=self.semester)
+
+        semester_overview_page = self.app.get(
+            reverse("staff:semester_view", args=[self.semester.pk]), user=self.manager, status=200
+        )
+
+        form = semester_overview_page.forms["evaluation_operation_form"]
+
+        form["evaluation"] = [evaluation.pk]
+
+        confirmation_page = form.submit(name="target_state", value=str(Evaluation.State.NEW))
+
+        if success_expected:
+            confirmation_page.forms["evaluation-operation-form"].submit()
+            self.assertEqual(Evaluation.objects.get(pk=evaluation.pk).state, Evaluation.State.NEW)
+        else:
+            self.assertNotIn("evaluation-operation-form", confirmation_page.forms)
```

Comment:
To me, it would be natural to assert that the evaluation state hasn't changed in the else case -- is there a reason you didn't do that?

At the same time, the fact that the form doesn't exist anymore seems more like a accidental implementation detail (I guess we redirect somewhere?) that might change.

Reply:
I replaced the check for the nonexisting form with a comparison of the HTTP Status code and assert that must not be 200. Or is it better to not do that check?